x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ssl/SSLServiceTests.java

```diff
@@ -681,7 +681,7 @@ public void testReadCertificateInformation() throws Exception {
         assertThat(cert.path(), equalTo(pemPath.toString()));
         assertThat(cert.format(), equalTo("PEM"));
         assertThat(cert.serialNumber(), equalTo("580db8ad52bb168a4080e1df122a3f56"));
-        assertThat(cert.subjectDn(), equalTo("CN=ad-ELASTICSEARCHAD-CA, DC=ad, DC=test, DC=elasticsearch, DC=com"));
+        assertThat(cert.subjectDn(), equalTo("CN=ad-ELASTICSEARCHAD-CA,DC=ad,DC=test,DC=elasticsearch,DC=com"));
```

Comment:
This one is a little tricky because technically it could be considered a breaking change to `CertificateInfo.subjectDn`, i.e., whereas we got a delimiter of ", " before we now lose the spaces because `getSubjectX500Principal` canonicalizes the distinguished name.

I don't think this is an issue (in fact this is part of why `getSubjectDN` is getting deprecated to begin with) but wanted to sanity check since I'm lacking context here. 

For what it's worth, we don't use `subjectDn`anywhere other than in this test suite to begin with.

Wdyt about just going ahead with this change?

Reply:
Good find. The `subjectDn` does make it to the REST layer with the [SSL certificate API](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-ssl.html). So strictly speaking, this is a breaking change. I think we should avoid it in this PR at least so it does not get blocked just having to justify this. IIUC, the existing behaviour can be achieved if we call `certificate.getSubjectX500Principal().toString()` instead of `...getName()`.

Reply:
Nice catch! I'm on board with using `toString()` here. Thanks for the suggestion.